src/Content.php

```diff
@@ -332,6 +371,30 @@ public function setValue($key, $value)
             }
         }
 
+        if ($key == 'templatefields') {
+            $oldValue = $this->values[$key];
+            if ((is_string($value)) || (is_array($value))) {
+                if (is_string($value)) {
+                    try {
+                        $unserdata = Lib::smartUnserialize($value);
+                    } catch (\Exception $e) {
+                        $unserdata = false;
+                    }
+                } else {
+                    $unserdata = $value;
+                }
+
+                if ($unserdata !== false) {
+                    $templateContent = new Content($this->app, $this->getTemplateFieldsContentType(), array(), false);
+                    $value = $templateContent;
+                    $this->populateTemplateFieldsContenttype($value);
```

Comment:
I'm not too sure...

From memory I *think* that this was from one of my older implementations and was superseded by the whole `$isRootType` thing. I haven't seen this throwing any errors yet... This should have come up by now, shouldn't it? Especially as there are tests in place for template fields.

I think it would be safe to remove it.